tests/test_cliPluginLoader.py

```diff
@@ -59,19 +65,96 @@ def cli():
     pass
 
 
+class NoLocalCommandsTestCLI(butler.LoaderCLI):
+    """Overrides LodaerCLI._getLocalCommands to not return local commands, so
+    local commands can be found via the DAF_BUTLER_PLUGINS environment var set
+    in the test via mock.patch.."""
+
+    @staticmethod
+    def _getLocalCommands():
+        """Mask the commands offered by daf_butler.
+
+        Returns
+        -------
+        dict of command name to list of packages name.
+            Key is the command name, value is the package that contains the
+            command.
+        """
+        return {}
+
+
 class Suite(unittest.TestCase):
 
     def test_loadAndExecuteCommand(self):
+        """Test that a command can be loaded and executed."""
         runner = click.testing.CliRunner()
         result = runner.invoke(cli, "command-test")
         self.assertEqual(result.exit_code, 0, result.output)
         self.assertEqual(result.stdout, "test command\n")
 
     def test_loadTopHelp(self):
+        """Test that an expected command is produced by 'butler --help'"""
         runner = click.testing.CliRunner()
         result = runner.invoke(cli, "--help")
         self.assertEqual(result.exit_code, 0, result.stdout)
-        self.assertIn("Commands:\n  command-test", result.stdout)
+        self.assertIn("command-test", result.stdout)
+
+    def test_getLocalCommands(self):
+        """Test getting the daf_butler CLI commands."""
+        localCommands = butler.LoaderCLI._getLocalCommands()
+        self.assertEqual(localCommands, {"config-dump": ["lsst.daf.butler.cli.cmd"],
```

Comment:
Every time you add a new daf_butler subcommand I think you are going to have to edit multiple tests here. It might help you to define the command list once and then create this dict from that.